aten/src/ATen/core/Generator.h

```diff
@@ -49,45 +50,56 @@
 
 namespace at {
 
-// The default seed is selected to be a large number
-// with good distribution of 0s and 1s in bit representation
-constexpr uint64_t default_rng_seed_val = 67280421310721;
-
 struct CAFFE2_API Generator {
-  // Constructors
-  Generator(Device device_in, DispatchKeySet key_set);
+  Generator() {}
 
-  // Delete all copy and move assignment in favor of clone()
-  // method
-  Generator(const Generator& other) = delete;
-  Generator(Generator&& other) = delete;
-  Generator& operator=(const Generator& other) = delete;
+  explicit Generator(std::shared_ptr<c10::GeneratorImpl> gen_impl)
+   : impl_(std::move(gen_impl)) {
+    if (impl_.get() == nullptr) {
+      throw std::runtime_error("GeneratorImpl with nullptr is not supported");
+    }
+  }
 
-  virtual ~Generator() = default;
-  std::shared_ptr<Generator> clone() const;
+  // TODO(pbelevich): delete this after replace Generator generator = nullptr with {}
+  Generator(c10::GeneratorImpl* gen_impl)
+   : impl_(std::shared_ptr<c10::GeneratorImpl>(gen_impl)) {}
```

Comment:
A much safer variant of this overload is `Generator(nullptr_t gen_impl)` which will only permit `nullptr` exactly.

Reply:
(If I read the TODO correctly, that is!)

Reply:
addressed